### DISCUSSION TITLE
Multi-GPU Training GPU Usage

### DISCUSSION THREAD

**Discussion by zhenhuahu** [Lightning-AI/pytorch-lightning#2701] [DDP / multi-GPU / multi-node] [answered]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### Hi, I'm using lightning and ddp as backend to do multi-gpu training, with Apex amp (amp_level = 'O1'). The gpu number is 8. I noticed that during training, most of time GPU0's utilization is 0%, while others are almost 100%. But their memory usage are the same. Is this normal? I use OpenPAI and have attached their utilization and memeory usage below. Thanks.

![amp_O1_gpu_usage](https://user-images.githubusercontent.com/11988890/88447203-9c8b8d00-cdfe-11ea-8a5b-b1c2551fc4e5.png)

#### What's your environment?

 - OS: [e.g. iOS, Linux, Win]
 - Packaging [e.g. pip, conda]
 - Version [e.g. 0.5.2.1]

**Comment by awaelchli** [comment]
check the cpu usage, to make sure dataloading is not a bottleneck.

**Comment by zhenhuahu** [comment]
> check the cpu usage, to make sure dataloading is not a bottleneck.

Hi, thanks for the reply. The total metric(including cpu, gpu) is as follows:

![Screenshot from 2020-07-26 00-58-22](https://user-images.githubusercontent.com/11988890/88471811-5b1cdf80-cedb-11ea-8b04-d71b5b4ef9b5.png)

From these images, I have no idea if dataloading is a bottleneck.

I also did profilers, as mentioned [in this issue](https://github.com/PyTorchLightning/pytorch-lightning/issues/2699#issuecomment-663932393) . The dataloading time is constant. But there's some weird call to ''torch._C._TensorBase' objects' that takes a lot of time.

Thanks!

**Comment by awaelchli** [comment, marked as answer]
Your cpu usage seems high. It could be the cpu is the bottleneck here. Try fewer gpus and observe then observe the gpu utilization.

**Comment by zhenhuahu** [comment]
> Your cpu usage seems high. It could be the cpu is the bottleneck here. Try fewer gpus and observe then observe the gpu utilization.

Yes. I have found it out. Thank you so much!
